types/three/three-core.d.ts

```diff
@@ -2075,7 +2087,7 @@ export class Loader {
 export interface LoaderHandler {
     handlers: (RegExp | Loader)[];
 
-    add(regex: RegExp, loader: Loader): void;
+    add(regex: RegExp, loader: Loader | CompressedTextureLoader): void;
```

Comment:
This smelled a little odd, so I did some digging. Here's what I went through trying to understand the `Loader.Handlers` system within three:
-  `Loader` is supposedly the base class for loaders. Why would a `CompressedTextureLoader` not already be a `Loader`?
- I can see that the js itself doesn't formally extend `Loader` when implementing `CompressedTextureLoader`, so this weirdness somewhat stems from the underlying JS.
- It seems that in practice the returned interface assumed by the callers of `Loader.Handlers.get` is actually different to the `Loader` base class.
- Yes, after reading more, it seems THREE only ever uses Handlers to store and retrieve texture loaders, not arbitrary loaders...
- Callers of `Handlers.get` also seem to assume the existence of `setCrossOrigin`, which CompressedTextureLoader doesn't have, and NOR DOES `Loader` ...
- I haven't looked at all outside of three.js to see how others are making use of the Handlers system, and what they're assuming.

I feel like the best we can say about `Loader.Handlers` is that its a lookup table of objects whose interfaces can vary but are all associated with loading stuff... They probably have a `load` method...

Well, that's all very well, but what shall we actually do here?

I propose we add an `AnyLoader` interface something like:
```
export interface AnyLoader {
  load(url: string, onLoad?: (result: any) => void, onProgress?: (event: ProgressEvent) => void, onError?: (event: ErrorEvent) => void): any;
}
```
And declare that `LoaderHandler` `add`s and `get`s `AnyLoader`s. `CompressedTextureLoader` and all the others should be compatible with that at least.

That way the type declarations avoid promising anything that isn't true, and we represent the `Handlers` system's full (terrifying) flexibility.

What do you think?